pkg/didcomm/crypto/crypter.go

```diff
@@ -18,21 +14,23 @@ type Crypter interface {
 	// returns:
 	// 		[]byte containing the encrypted envelope
 	//		error if encryption failed
-	Encrypt(payload []byte, sender KeyPair, recipients []*[chacha.KeySize]byte) ([]byte, error)
+	// TODO add key type of recipients and sender keys to be validated by the implementation
```

Comment:
More than validated - we should have a higher-level crypto that switches based on the algorithm.

Reply:
currently this can be checked in the payload for decryption

but yes, a wrapper of this interface should be able to switch implementations based on algorithm

Reply:
I'll update the comment..